### ISSUE TITLE
dist: build on ubuntu 14.04 failing

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#1061]
http://jenkins.cloudius-systems.com:8080/job/scylla-ubuntu-deb/28/

```
[46/49] CXX build/release/http/reply.o
[47/49] CXX build/release/core/scollectd.o
[48/49] LINK build/release/apps/iotune/iotune
[49/49] AR build/release/libseastar.a
ninja: build stopped: subcommand failed.
make[1]: *** [override_dh_auto_build] Error 1
make[1]: Leaving directory `<http://jenkins.cloudius-systems.com:8080/job/scylla-ubuntu-deb/ws/scylla'>
make: *** [binary] Error 2
dpkg-buildpackage: error: fakeroot debian/rules binary gave error exit status 2
debuild: fatal error at line 1364:
dpkg-buildpackage -rfakeroot -D -us -uc failed
```

**Comment by slivne** [comment]
build http://jenkins.cloudius-systems.com:8080/job/scylla-ubuntu-deb/27/ also failed

build http://jenkins.cloudius-systems.com:8080/job/scylla-ubuntu-deb/26 passed (6a3872b35598321006c60151609b8213ec01eee5)

**Comment by avikivity** [comment]
Looks like somehow there was a file `seastar/build/release/gen`, and `mkdir` was not able to cope with it.

**Comment by slivne** [comment]
@avikivity  can you send a patch ?

**Comment by avikivity** [comment]
It's not clear where the problem is.  Did you wipe out the workspace?

**Comment by slivne** [comment]
ubuntu is built of a machine booted on AWS - so the workspace is erased
once the machine is shutdown -you can trigger a build and then connect to
the machine

**Comment by syuu1228** [comment]
@slivne I'll handle this

**Comment by syuu1228** [comment]
I found following build error on http_response_parser.hh:

```
[39/362] CXX build/release/gen/cassandra_constants.o
[40/362] CXX build/release/gen/Cassandra.o
[41/362] ANTLR3 cql3/Cql.g
[42/362] NINJA seastar/build/release/libseastar.a seastar/build/release/apps/iotune/iotune
[43/362] GEN seastar/http/request_parser.hh
[44/362] GEN seastar/http/http_response_parser.hh
FAILED: /usr/bin/ninja -C seastar build/release/gen/http/http_response_parser.hh build/debug/gen/http/http_response_parser.hh
ninja: Entering directory `seastar'
[1/2] RAGEL build/release/gen/http/http_response_parser.hh
ninja: error: mkdir(build/release/gen): File exists
ninja: build stopped: .
ninja: Entering directory `seastar'
[1/2] RAGEL build/debug/gen/http/request_parser.hh
[2/2] RAGEL build/release/gen/http/request_parser.hh
ninja: Entering directory `seastar'
[1/49] make -C build/dpdk
[2/49] RAGEL build/release/gen/http/request_parser.hh
[3/49] RAGEL build/release/gen/apps/memcached/ascii.hh
[4/49] RAGEL build/release/gen/http/http_response_parser.hh
[5/49] SWAGGER build/release/gen/apps/httpd/demo.json.hh
```

**Comment by penberg** [comment]
@avikivity, I added the following commit to ensure ``http_response_parser.hh`` is generated early enough: 97bef4fb7c05381d05e9c55551fa1a570db03ca4. Can you see anything wrong with the patch?

**Comment by penberg** [comment]
The ``exist_ok`` argument in ``os.makedirs`` only exists in Python 3.2 and later. I think Ubuntu 14.04 still uses Python 2. Does anyone have access to the machine to confirm that?

**Comment by avikivity** [comment]
There should be only one ninja call to seastar, parallel calls can cause corruption (even if generating different files).

**Comment by avikivity** [comment]
docker run -i -t --rm ubuntu:14.04 /bin/bash

**Comment by avikivity** [comment]
Appparently it does has Python 3.4.3.

**Comment by penberg** [comment]
I sent a patch to fix parallel invocations of seastar build. I was not able to reproduce the issue in a Docker container, though.

**Comment by penberg** [comment]
@slivne @syuu1228 This issue is likely fixed by 4892a6ded94790f4db65229cd8edcac56c3c9805 but as I was not able to reproduce, I think we should lower the priority to "high" and wait for a while to see if it happens again on the Jenkins build.
